python-sdk/src/astro/databases/base.py

```diff
@@ -314,7 +315,7 @@ def create_table(
         """
         if table.columns:
             self.create_table_using_columns(table)
-        elif file and self.is_native_autodetect_schema_available(file):
+        elif use_native_support and file and self.is_native_autodetect_schema_available(file):
             self.create_table_using_native_schema_autodetection(table, file)
         else:
```

Comment:
What happens in the following case:

the user defines:
* `table.columns`
* `use_native_support`

And `is_native_autodetect_schema_available` is `True`.

Would we both call:
* `self.create_table_using_columns(table)`
* `self.create_table_using_native_schema_autodetection(table, file)`

Reply:
Yeah that makes sense. If a user wants to explicitly state the columns we shouldn't need to look anything up. If they are incorrect in their definition then it's on them to fix it.